sceptre/stack.py

```diff
@@ -194,6 +194,7 @@ def __init__(
         ignore=False,
         obsolete=False,
         stack_group_config: dict = {},
+        config: dict = {},
```

Comment:
It's a very bad practice to have mutable default arguments in Python. I know `stack_group_config` is doing it and that shouldn't be doing it either. The better approach is to set the value to `None` and then do like you're doing below in line 215.